File: frontend/pubbyapp/src/app/model/question.ts
import { PossibleAnswer } from "./possible-answer";

export class Question {
	public questionId?: number;
	public title?: string;
	public questionText?: string;
	public possibleAnswers: PossibleAnswer[] = [];


	constructor(questionId?: number, title?: string, questionText?: string, possibleAnswers: PossibleAnswer[] = []) {
		this.questionId = questionId;
		this.title = title;
		this.questionText = questionText;
		this.possibleAnswers = possibleAnswers;
	}
}

File: src/main/java/com/vada/models/Question.java
package com.vada.models;

import java.util.List;

import org.springframework.stereotype.Component;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.vada.tools.PossibleAnswerListSerializer;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.Data;

@Data
@Entity
@Component
@Table(name = "questions")
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({ "questionId", "title", "questionText", "possibleAnswers" })
public class Question {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long questionId;
    @Column(name = "title")
    private String title;
    @Column(name = "questionText")
    private String questionText;
    @JsonSerialize(using = PossibleAnswerListSerializer.class)
    @OneToMany(mappedBy = "question", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    @JsonManagedReference
    private List<PossibleAnswer> possibleAnswers;

}
